core/src/main/java/apoc/convert/Json.java

```diff
@@ -1,21 +1,76 @@
 package apoc.convert;
 
+import apoc.meta.Meta;
 import apoc.result.MapResult;
 import apoc.util.JsonUtil;
 import apoc.util.Util;
-import org.neo4j.graphdb.Entity;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.Path;
-import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.*;
 import org.neo4j.procedure.*;
 
 import java.io.IOException;
 import java.util.*;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static apoc.util.Util.labelStrings;
+import static apoc.util.Util.map;
 
 public class Json {
 
+    private Object writeJsonResult(Object value) {
+        Meta.Types type = Meta.Types.of(value);
+        switch (type) {
+            case NODE:
+                return nodeToMap((Node) value, true);
+            case RELATIONSHIP:
+                return relToMap((Relationship) value);
+            case PATH:
+                return writeJsonResult(StreamSupport.stream(((Path)value).spliterator(),false)
+                        .map(i-> i instanceof Node ? nodeToMap((Node) i, true) : relToMap((Relationship) i))
+                        .collect(Collectors.toList()));
+            case LIST:
+                return ((ArrayList) value).stream().map(this::writeJsonResult).collect(Collectors.toList());
```

Comment:
occhio potrebbe essere anche un array semplice, non necessariamente un arraylist
```suggestion
                
                return Convert.convertToList(value).stream().map(this::writeJsonResult).collect(Collectors.toList());
```